docs/resources/access_control_rule_set.md

```diff
@@ -0,0 +1,170 @@
+
+This resource allows you to manage access rules on Databricks account level resources. For convenience we allow accessing this resource through the Databricks account and workspace.
+
+-> **Note** Currently, we only support managing access rules on service principal resources through databricks_access_control_rule_set.
+
+-> **Warning** databricks_access_control_rule_set cannot be used to manage access rules for resources supported by [databricks_permissions](permissions.md). Refer to its documentation for more information.
+
+## Example usage
+
+Rule set management through a Databricks workspace:
+
+```hcl
+locals {
+  account_id = "00000000-0000-0000-0000-000000000000"
+}
+
+// account level group
+data "databricks_group" "ds" {
+  display_name = "Data Science"
+}
+
+resource "databricks_service_principal" "automation_sp" {
+  display_name = "SP_FOR_AUTOMATION"
+}
+
+resource "databricks_access_control_rule_set" "automation_sp_rule_set" {
+  name = "accounts/${local.account_id}/servicePrincipals/${databricks_service_principal.automation_sp.application_id}/ruleSets/default"
+
+  grant_rules {
+    principals = ["groups/${data.databricks_group.ds.display_name}"]
+    role       = "roles/servicePrincipal.user"
+  }
+}
+```
+
+Rule set management through AWS Databricks account:
+
+```hcl
+// initialize provider at account-level
+provider "databricks" {
+  alias      = "mws"
+  host       = "https://accounts.cloud.databricks.com"
+  account_id = "00000000-0000-0000-0000-000000000000"
+  username   = var.databricks_account_username
+  password   = var.databricks_account_password
+}
+
+// account level group creation
+resource "databricks_group" "ds" {
+  display_name = "Data Science"
+}
+
+resource "databricks_service_principal" "automation_sp" {
+  display_name = "SP_FOR_AUTOMATION"
+}
+
+resource "databricks_access_control_rule_set" "automation_sp_rule_set" {
+  name = "accounts/${provider.databricks.account_id}/servicePrincipals/${databricks_service_principal.automation_sp.application_id}/ruleSets/default"
+
+  grant_rules {
+    principals = ["groups/${databricks_group.ds.display_name}"]
+    role       = "roles/servicePrincipal.user"
+  }
+}
+```
+
+Rule set management through Azure Databricks account:
+
+```hcl
+// initialize provider at Azure account-level
+provider "databricks" {
+  alias      = "azure_account"
+  host       = "https://accounts.azuredatabricks.net"
+  account_id = "00000000-0000-0000-0000-000000000000"
+  auth_type  = "azure-cli"
+}
+
+// account level group creation
+resource "databricks_group" "ds" {
+  display_name = "Data Science"
+}
+
+resource "databricks_service_principal" "automation_sp" {
+  application_id = "00000000-0000-0000-0000-000000000000"
+  display_name   = "SP_FOR_AUTOMATION"
+}
+
+resource "databricks_access_control_rule_set" "automation_sp_rule_set" {
+  name = "accounts/${provider.databricks.account_id}/servicePrincipals/${databricks_service_principal.automation_sp.application_id}/ruleSets/default"
```

Comment:
this `${provider.databricks.account_id}` interpolation is not allowed by Terraform, you'll need to factor out `account_id` as a variable/local